main.go

```diff
@@ -154,19 +171,22 @@ func initDevice() (*streamdeck.Device, error) {
 	}
 	ver, err := dev.FirmwareVersion()
 	if err != nil {
+		closeDevice(&dev)
```

Comment:
I wonder if we shouldn't just try to reset the device once after opening it? After `closeDevice` we're terminating deckmaster anyway, as we return an error below.

Reply:
Understood, but instead of closing the device and then shutting down again, maybe we could just try to close the device and re-open it inside Deckmaster, if we encounter an error during initialization? That would prevent the user from having to re-plug the device *or* restart Deckmaster. What do you think?

Reply:
This might not be what you're getting at, but in my case, when the device is in a bad state (pretty much when `dev.Close()` is not called), it stays in that bad state until I re-plug the device.

In using deckmaster so far (which is only for a few days), those initialization operations (get firmware version, reset, set brightness) have never failed, but that doesn't mean they won't ever fail.  What I'm trying to say is that I probably don't care too much how this is handled here, but at least we should `dev.Close()` so that there's less chance of needed a re-plug at this point.

As for a retry of initialization, I suppose that's possible, but I personally don't think it's necessary.  At initialization time, it would be pretty obvious if things are not working, and the user could start troubleshooting.  If it's hardware failure, we probably don't want to keep retrying "known good" things that are suddenly failing.

I've pushed the change modeled on what I said above, but I'll change it to whatever you feel is best here since I don't feel too strongly about how things fail at initialization time.

Reply:
Thank you, starting to understand the situation a little better now!